pandas/tests/io/parser/test_network.py

```diff
@@ -50,151 +93,142 @@ def check_compressed_urls(salaries_table, compression, extension, mode,
     tm.assert_frame_equal(url_table, salaries_table)
 
 
-class TestS3(object):
+@tm.network
+def test_parse_public_s3_bucket():
+    pytest.importorskip('s3fs')
+    # more of an integration test due to the not-public contents portion
+    # can probably mock this though.
+    for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
+        df = read_csv('s3://pandas-test/tips.csv' +
+                      ext, compression=comp)
+        assert isinstance(df, DataFrame)
+        assert not df.empty
+        tm.assert_frame_equal(read_csv(
+            tm.get_data_path('tips.csv')), df)
+
+    # Read public file from bucket with not-public contents
+    df = read_csv('s3://cant_get_it/tips.csv')
+    assert isinstance(df, DataFrame)
+    assert not df.empty
+    tm.assert_frame_equal(read_csv(tm.get_data_path('tips.csv')), df)
+
+
+def test_parse_public_s3n_bucket(test_s3_resource):
+
+    # Read from AWS s3 as "s3n" URL
+    df = read_csv('s3n://pandas-test/tips.csv', nrows=10)
+    assert isinstance(df, DataFrame)
+    assert not df.empty
+    tm.assert_frame_equal(read_csv(
+        tm.get_data_path('tips.csv')).iloc[:10], df)
 
-    def setup_method(self, method):
-        try:
-            import s3fs  # noqa
-        except ImportError:
-            pytest.skip("s3fs not installed")
 
-    @tm.network
-    def test_parse_public_s3_bucket(self):
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df = read_csv('s3://pandas-test/tips.csv' +
-                          ext, compression=comp)
+def test_parse_public_s3a_bucket(test_s3_resource):
+    # Read from AWS s3 as "s3a" URL
+    df = read_csv('s3a://pandas-test/tips.csv', nrows=10)
+    assert isinstance(df, DataFrame)
+    assert not df.empty
+    tm.assert_frame_equal(read_csv(
+        tm.get_data_path('tips.csv')).iloc[:10], df)
+
+
+def test_parse_public_s3_bucket_nrows(test_s3_resource):
```

Comment:
since u have so many s3 tests would be ok with making a new file just for them (ok leaving them here too)